optaplanner-core/src/main/java/org/optaplanner/core/api/solver/ConsumptionPause.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.api.solver;
+
+/**
+ * Represents a pause of a solution consumer initiated by calling
+ * {@link SolverManager#pauseBestSolutionConsumer(Object)}.
+ */
+public interface ConsumptionPause extends AutoCloseable {
+
+    @Override
+    void close();
```

Comment:
Overriding the `close()` method as otherwise, `AutoCloseable` enforces a user to catch `Exception` in the try-with-resources block.